src/Messages/DTO/MessagePart.php

```
@@ -0,0 +1,264 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Messages\DTO;
+
+use WordPress\AiClient\Common\Contracts\WithJsonSchema;
+use WordPress\AiClient\Files\DTO\InlineFile;
+use WordPress\AiClient\Files\DTO\RemoteFile;
+use WordPress\AiClient\Messages\Enums\MessagePartTypeEnum;
+use WordPress\AiClient\Tools\DTO\FunctionCall;
+use WordPress\AiClient\Tools\DTO\FunctionResponse;
+
+/**
+ * Represents a part of a message.
+ *
+ * Messages can contain multiple parts of different types, such as text, files,
+ * function calls, etc. This DTO encapsulates one such part.
+ *
+ * @since n.e.x.t
+ */
+class MessagePart implements WithJsonSchema
+{
+    /**
+     * @var MessagePartTypeEnum The type of this message part.
+     */
+    private MessagePartTypeEnum $type;
+
+    /**
+     * @var string|null Text content (when type is TEXT).
+     */
+    private ?string $text = null;
+
+    /**
+     * @var InlineFile|null Inline file data (when type is INLINE_FILE).
+     */
+    private ?InlineFile $inlineFile = null;
+
+    /**
+     * @var RemoteFile|null Remote file reference (when type is REMOTE_FILE).
+     */
+    private ?RemoteFile $remoteFile = null;
+
+    /**
+     * @var FunctionCall|null Function call request (when type is FUNCTION_CALL).
+     */
+    private ?FunctionCall $functionCall = null;
+
+    /**
+     * @var FunctionResponse|null Function response (when type is FUNCTION_RESPONSE).
+     */
+    private ?FunctionResponse $functionResponse = null;
+
+    /**
+     * Private constructor to enforce factory method usage.
+     *
+     * @since n.e.x.t
+     *
+     * @param MessagePartTypeEnum $type The type of this message part.
+     */
+    private function __construct(MessagePartTypeEnum $type)
+    {
+        $this->type = $type;
```

See my other comment, instead of passing in the type here, we should allow passing in the actual data structure, and then the class can automatically determine the type value based on that. That way the class works equivalent to the other DTOs, allowing to pass in all necessary data in the constructor.

So basically replace the parameter with something like `$data`, which can be either a `string`, `InlineFile`, `RemoteFile`, etc. (validation would need to happen inside the constructor, since we can't use union types). We can throw an `InvalidArgumentException` if it's neither of them.

Then, based on what `$data` is, set it in the respective class property, and set the `$type` property accordingly.

---

Resolved in eb386cf